### PR TITLE
Updated qlearning.py

### PR DESCRIPTION
**Pull request by Hasaanmaqsood**
Updated qlearning.py to add training & evaluation plots and save agent simulation as MP4

### CODE REVIEW

File: src/agents/tabular/qlearning.py [hunk 1]
```diff
@@ -1,72 +1,205 @@
...
                 )
 
-                # Transition to the next state
                 state = next_state
+                total_reward += reward
+
+            # Store total reward
+            self.train_rewards_log.append(total_reward)
+
+            # Early stopping check
```
Comment (daniil-shlenskii) [inline src/agents/tabular/qlearning.py:85]: Early stopping should not be used with a metric collected on training data.
It results in overfitting.

Metric for early stopping should be calculated on the validation data instead.
In our case it is a trajectory of a current step fixed policy
